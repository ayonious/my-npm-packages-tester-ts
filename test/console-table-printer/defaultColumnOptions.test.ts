import { expect } from '@jest/globals';
import { Table } from 'console-table-printer';

describe('Table with defaultColumnOptions', () => {
  it('should apply default column options to all columns', () => {
    const table = new Table({
      defaultColumnOptions: {
        alignment: 'right',
        color: 'green'
      },
      columns: [
        { name: 'id' },
        { name: 'name' },
        { name: 'value' }
      ]
    });

    table.addRow({ id: 1, name: 'Item 1', value: 100 });
    table.addRow({ id: 2, name: 'Item 2', value: 200 });

    expect(table.render()).toMatchSnapshot();
  });

  it('should allow column-specific options to override defaults', () => {
    const table = new Table({
      defaultColumnOptions: {
        alignment: 'right',
        color: 'green'
      },
      columns: [
        { name: 'id', alignment: 'left' },
        { name: 'name', color: 'blue' },
        { name: 'value' }
      ]
    });

    table.addRow({ id: 1, name: 'Item 1', value: 100 });
    table.addRow({ id: 2, name: 'Item 2', value: 200 });

    expect(table.render()).toMatchSnapshot();
  });

  it('should work with no column definitions', () => {
    const table = new Table({
      defaultColumnOptions: {
        alignment: 'right',
        color: 'yellow'
      }
    });

    table.addRow({ id: 1, name: 'Item 1', value: 100 });
    table.addRow({ id: 2, name: 'Item 2', value: 200 });

    expect(table.render()).toMatchSnapshot();
  });
});
