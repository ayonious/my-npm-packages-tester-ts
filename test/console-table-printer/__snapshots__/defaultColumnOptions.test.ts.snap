// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Table with defaultColumnOptions should allow column-specific options to override defaults 1`] = `
"┌────┬────────┬───────┐
│[37m [0m[01mid[0m │[37m [0m[01m  name[0m │[37m [0m[01mvalue[0m │
├────┼────────┼───────┤
│[37m [0m[32m1 [0m │[37m [0m[34mItem 1[0m │[37m [0m[32m  100[0m │
│[37m [0m[32m2 [0m │[37m [0m[34mItem 2[0m │[37m [0m[32m  200[0m │
└────┴────────┴───────┘"
`;

exports[`Table with defaultColumnOptions should apply default column options to all columns 1`] = `
"┌────┬────────┬───────┐
│[37m [0m[01mid[0m │[37m [0m[01m  name[0m │[37m [0m[01mvalue[0m │
├────┼────────┼───────┤
│[37m [0m[32m 1[0m │[37m [0m[32mItem 1[0m │[37m [0m[32m  100[0m │
│[37m [0m[32m 2[0m │[37m [0m[32mItem 2[0m │[37m [0m[32m  200[0m │
└────┴────────┴───────┘"
`;

exports[`Table with defaultColumnOptions should work with no column definitions 1`] = `
"┌────┬────────┬───────┐
│[37m [0m[01mid[0m │[37m [0m[01m  name[0m │[37m [0m[01mvalue[0m │
├────┼────────┼───────┤
│[37m [0m[33m 1[0m │[37m [0m[33mItem 1[0m │[37m [0m[33m  100[0m │
│[37m [0m[33m 2[0m │[37m [0m[33mItem 2[0m │[37m [0m[33m  200[0m │
└────┴────────┴───────┘"
`;
